lib/logstash/filters/dns.rb

```diff
@@ -122,6 +122,12 @@ def resolve(event)
     @resolve.each do |field|
       is_array = false
       raw = event.get(field)
+
+      if raw.nil?
+        @logger.warn("DNS filter could not resolve missing field", :field => field)
```

Comment:
How is the number of fields variable? You have to explicitly provide them right?

If your config is:

```
dns {
  resolve => [fieldA, fieldB]
}
```

You can easily port that to:

```
if ([fieldA] && [fieldB]) {
  dns { resolve => [fieldA, fieldB] }
}
```